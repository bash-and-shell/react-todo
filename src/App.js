import './App.scss';
import { Container, Tab, Tabs } from 'react-bootstrap';
import AddTask from './components/AddTask.js';
import "bootstrap/dist/css/bootstrap.css";
import Task from './components/Task.js';
import TaskList from './components/TaskList';
import NewList from './components/NewList';
import { useState } from 'react';


const App = () => {

  const [tasks, setTasks] = useState([])
  const [tabKey, setTabKey] = useState('all')

  const addTask = (task) => {
    setTasks([...tasks, task])
  }

  const deleteTask = (task) => {
    const updateTasks = [...tasks]
    updateTasks.splice(task, 1)
    setTasks(updateTasks)
  }

  const editTask = (task) => {
    const edit = prompt('Edit task:');
    if (edit !== null && edit.trim() !== '') {
      let updateTasks = [...tasks]
      updateTasks[task] = edit
      setTasks(updateTasks)
    }
  }

  const addList = () => {
    
  }

  return (
    <div className="App">
      <Container component="main" maxWidth="xs">
        <AddTask onSubmit={addTask} />
        <NewList onSubmit={(e) => {console.log(e)}} />
        <Tabs
          id="tab-list"
          activeKey={tabKey}
          onSelect={k => setTabKey(k)}
        >
          <Tab eventKey='all' title='All'>
            <TaskList>
              {tasks.map((task, index) => {
                return <Task key={index} text={task} handleDelete={deleteTask} handleEdit={editTask} />
              })}
            </TaskList>
          </Tab>
        </Tabs>


      </Container>

    </div>
  );
}

export default App;
