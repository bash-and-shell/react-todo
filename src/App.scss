.App {
  text-align: center;

  .row {
    padding: 0.5rem
  }
}
