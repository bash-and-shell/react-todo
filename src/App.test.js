import renderer from 'react-test-renderer'
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import App from './App';

test('matches snapshot', () => {
  const component = renderer.create(<App />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('adds element', () => {
  render(<App />)

  fireEvent.change(screen.getByPlaceholderText("Enter task"), { target: {value: 'hello'} })
  fireEvent.click(screen.getByText("Add"))

  expect(screen.getByText('hello'))
})

test('deletes element', () => {
  render(<App />)

  fireEvent.change(screen.getByPlaceholderText("Enter task"), { target: {value: 'hello'} })
  fireEvent.click(screen.getByText("Add"))

  const item = screen.getByText('hello')
  expect(item)

  fireEvent.click(screen.getByText("Delete"))

  expect(item).not.toBeInTheDocument()
})

