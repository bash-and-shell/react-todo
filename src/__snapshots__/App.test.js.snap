// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="container"
    component="main"
    maxWidth="xs"
  >
    <form
      className="add-task"
      onSubmit={[Function]}
    >
      <label
        className="form-label"
      >
        Enter New Task
      </label>
      <div
        className="add-task-group"
      >
        <input
          className="form-control"
          id="addTask"
          onChange={[Function]}
          placeholder="Enter task"
          type="text"
          value=""
        />
        <button
          className="add-task-button btn btn-primary"
          disabled={false}
          type="submit"
        >
          Add
        </button>
      </div>
    </form>
    <div
      className="task-list list-group"
      onKeyDown={[Function]}
    />
  </div>
</div>
`;
